src/azul/deployment.py

```diff
@@ -482,12 +482,21 @@ def _log_client_response(self,
                              **kwargs: Any
                              ) -> Optional[AWSResponse]:
         event_name = self._shorten_event_name(event_name, self._response_event_name)
-        response = kwargs['response_dict']
-        boto3_log.info('%s:\tGot %s response', event_name, response['status_code'])
-        message = http_body_log_message('response',
-                                        response.get('body'),
-                                        verbatim=self._log_body_verbatim)
-        boto3_log.info('%s:\t%s', event_name, message)
+        response, exception = kwargs['response_dict'], kwargs['exception']
+        if exception is None:
+            if response is None:
+                boto3_log.info('%s:\tGot no response', event_name)
+            else:
+                boto3_log.info('%s:\tGot %s response', event_name, response['status_code'])
+                message = http_body_log_message('response',
+                                                response['body'],
+                                                verbatim=self._log_body_verbatim)
+                boto3_log.info('%s:\t%s', event_name, message)
+        else:
+            if response is None:
+                pass  # assuming that exception will be raised and lead to a backtrace in the logs
+            else:
+                assert False, (exception, response)
```

Comment:
```suggestion
                assert False, (exception, type(response))
```

Could be big/unprintable.